pkg/tarutil/tarutil.go

```diff
@@ -164,17 +165,29 @@ func ExtractFiles(r io.Reader, filenameMatcher matcher.Matcher) (LayerFiles, err
 		case tar.TypeReg, tar.TypeLink:
 			var fileData FileData
 
+			executable, _ := executableMatcher.Match(filename, hdr.FileInfo(), contents)
 			if hdr.Size > maxELFExecutableFileSize {
 				log.Warnf("Skipping ELF executable check for file %q (%d bytes) because it is larger than the configured maxELFExecutableFileSizeMB of %d", filename, hdr.Size, maxELFExecutableFileSize/1024/1024)
-			} else if hdr.Size >= elfHeaderSize { // Only bother attempting to get ELF metadata if the file is large enough for the ELF header.
-				fileData.ELFMetadata, err = elf.GetExecutableMetadata(contents)
-				if err != nil {
-					log.Errorf("Failed to get dependencies for %s: %v", filename, err)
+			} else {
+				if hdr.Size >= elfHeaderSize { // Only bother attempting to get ELF metadata if the file is large enough for the ELF header.
+					fileData.ELFMetadata, err = elf.GetExecutableMetadata(contents)
+					if err != nil {
+						log.Errorf("Failed to get dependencies for %s: %v", filename, err)
+					}
+				}
+				if executable && fileData.ELFMetadata == nil {
```

Comment:
so if this file is smaller than `maxELFExecutableFileSize`, was determined to not be an ELF file (due to failure or it's just not an ELF file), and has the executable bit set, then we declare this file is executable iff it is larger than 2 bytes and has a shebang header?

Reply:
What if the script doesn't have the shebang header? Isn't that possible?

Reply:
@c-du Does this mean that we can't find the dependencies for this executable anyways if it is actually ELF?

Reply:
> What if the script doesn't have the shebang header? Isn't that possible?
This also means that the process will come in like `bash -c script` so it's different scenario altogether